MSAL/test/integration/native_auth/end_to_end/MSALNativeAuthEndToEndBaseTestCase.swift

```diff
@@ -26,54 +26,65 @@ import XCTest
+    private let codeRetriever = MSALNativeAuthEmailCodeRetriever()
 
     override func setUpWithError() throws {
         try super.setUpWithError()
-        let useMockAPIBooleanString = ProcessInfo.processInfo.environment["useMockAPI"] ?? "false"
-        usingMockAPI = Bool(useMockAPIBooleanString) ?? false
         
-        // mock API URL needs to contains a tenant
-        guard let authorityURL = URL(string: Configuration.authorityURLString + (usingMockAPI ? "/testTenant" : "")), let authority = try? MSALCIAMAuthority(url: authorityURL) else {
-            XCTFail("AuthorityURL not set or invalid")
+        guard let confURL = Bundle(for: Self.self).url(forResource: "conf", withExtension: "json"), let configurationData = try? Data(contentsOf: confURL) else {
+            XCTFail("conf.json file not found")
             return
         }
-
-        sut = try MSALNativeAuthPublicClientApplication(
-            configuration: MSALPublicClientApplicationConfig(
-                clientId: Configuration.clientId,
-                redirectUri: nil,
-                authority: authority
-            ),
-            challengeTypes: [.OOB, .password]
-        )
-
-        if usingMockAPI {
-            print("🤖🤖🤖 Using mock API: \(Configuration.authorityURLString)")
-        } else {
-            print("👩‍💻👩‍💻👩‍💻 Using test tenant: \(Configuration.authorityURLString)")
+        let confFile = try? JSONSerialization.jsonObject(with: configurationData, options: []) as? [String: Any]
+        confFileContent = confFile?[Constants.nativeAuthKey] as? [String: String]
+    }
+    
+    func initialisePublicClientApplication(
+        useEmailPasswordClientId: Bool = true,
+        attributeRequiredClientId: Bool = false,
+        challengeTypes: MSALNativeAuthChallengeTypes = [.OOB, .password]
+    ) -> MSALNativeAuthPublicClientApplication? {
+        var clientIdKey = Constants.clientIdEmailPasswordKey
+        if useEmailPasswordClientId && attributeRequiredClientId {
+            clientIdKey = Constants.clientIdEmailPasswordAttributesKey
+        } else if !useEmailPasswordClientId && attributeRequiredClientId {
+            clientIdKey = Constants.clientIdEmailCodeAttributesKey
+        } else if !useEmailPasswordClientId && !attributeRequiredClientId {
+            clientIdKey = Constants.clientIdEmailCodeKey
+        }
+        guard let clientId = confFileContent?[clientIdKey] as? String, let tenantSubdomain = confFileContent?[Constants.tenantSubdomainKey] as? String else {
+            XCTFail("ClientId or tenantSubdomain not found in conf.json")
+            return nil
         }
+        return try? MSALNativeAuthPublicClientApplication(clientId: clientId, tenantSubdomain: tenantSubdomain, challengeTypes: challengeTypes)
+    }
+    
+    func generateSignUpRandomEmail() -> String {
+        let randomId = UUID().uuidString.prefix(8)
+        return "native-auth-signup-\(randomId)@1secmail.org"
```

Comment:
Emails are with `.org` and API is with `.com`?

Reply:
Correct